client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js

```diff
@@ -36,9 +37,19 @@ class PaymentMethodStripe extends Component {
 		onCancel: PropTypes.func.isRequired,
 		onEditField: PropTypes.func.isRequired,
 		onDone: PropTypes.func.isRequired,
+		site: PropTypes.shape( {
+			title: PropTypes.string,
+		} ),
 	};
 
 	onEditFieldHandler = ( e ) => {
+		// Limit the statement descriptor field to 22 characters
+		if ( e.target && e.target.name && 'statement_descriptor' === e.target.name ) {
```

Comment:
I think that you can drop the middle check `e.target.name`. The next one will still work even if name is not defined.